replica/table.cc

```diff
@@ -438,14 +432,13 @@ compaction_group::do_add_sstable(lw_shared_ptr<sstables::sstable_set> sstables,
     if (backlog_tracker) {
         table::add_sstable_to_backlog_tracker(get_backlog_tracker(), sstable);
     }
-    // update sstable set last in case either updating
-    // staging sstables or backlog tracker throws
-    _t.update_stats_for_new_sstable(sstable->bytes_on_disk());
     return new_sstables;
 }
 
 void compaction_group::add_sstable(sstables::shared_sstable sstable) {
+    auto sstable_size = sstable->bytes_on_disk();
     _main_sstables = do_add_sstable(_main_sstables, std::move(sstable), enable_backlog_tracker::yes);
+    _main_set_disk_space_used += sstable_size;
```

Comment:
I don't see the code counterpart which subtracts the size of a deleted sstable.

Reply:
this is preexisting. there's no remove_sstable() procedure. when removing sstables (e.g. on compaction completion), we always call a procedure that rebuilds the list, which is never updated in-place anyway to support concurrent readers (even for add_sstable()). see update_main_sstable_list_on_compaction_completion().

Reply:
set_main_sstables() below is called when the main sstable set is rebuilt.

Reply:
I see.